core/types/header_ext.go

```diff
@@ -0,0 +1,197 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"io"
+	"math/big"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/common/hexutil"
+	ethtypes "github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/rlp"
+)
+
+// HeaderExtra is a struct that contains extra fields used by Avalanche
+// in the block header.
+// This type uses HeaderSerializable to encode and decode the extra fields
```

Comment:
nit for godoc automated links

```suggestion
// This type uses [HeaderSerializable] to encode and decode the extra fields
```